src/lib.rs

```diff
@@ -1,25 +1,45 @@
-use std::io::BufRead;
+use std::path::Path;
 
+use rayon::iter::ParallelBridge;
+use rayon::prelude::*;
+
+mod chunks;
 mod counter;
 mod key_finder;
 
+pub use chunks::chunks;
 pub use counter::{Counter, KeyCount};
 pub use key_finder::KeyFinder;
 
-pub fn top_few_from_stream<T: BufRead>(
-    reader: T,
+fn sum(mut l: Counter, r: Counter) -> Counter {
+    for (key, count) in r.counts().into_iter() {
+        l.add(&key, count);
+    }
+    l
+}
+
+pub fn top_few_from_stream(
+    path: &Path,
     kf: &KeyFinder,
     num: usize,
-) -> Result<Vec<KeyCount>, anyhow::Error> {
-    let mut counter = Counter::new(num);
-    let mut s = String::new();
-    for ln in reader.lines() {
-        s.clear();
-        let ln = ln?;
-        if let Ok(key) = kf.key(&ln, &mut s) {
-            counter.add(key)
-        }
-    }
+) -> anyhow::Result<Vec<KeyCount>> {
+    let total = chunks(path, 1 << 26)?
+        .collect::<Vec<_>>()
```

Comment:
I don't think it makes sense here to first `collect()` the iterator into a `Vec` and then immediately turn it into an `Iterator` again using `into_iter()`?

Reply:
It doesn't. I couldn't get this to work without collect.